openwisp_controller/geo/api/views.py

```diff
@@ -186,6 +187,61 @@ class GeoJsonLocationList(
     filterset_class = LocationOrganizationFilter
 
 
+class IndoorCoodinatesViewPagination(ListViewPagination):
+    page_size = 50
+
+
+class IndoorCoordinatesList(ProtectedAPIMixin, generics.ListAPIView):
+    serializer_class = IndoorCoordinatesSerializer
+    filter_backends = [filters.DjangoFilterBackend]
+    pagination_class = IndoorCoodinatesViewPagination
+    queryset = (
+        DeviceLocation.objects.filter(
+            location__type="indoor",
+            floorplan__isnull=False,
+        )
+        .select_related(
+            "content_object", "location", "floorplan", "content_object__organization"
+        )
+        .order_by("floorplan__floor")
+    )
+
+    def get_available_floors(self):
+        location_id = self.kwargs.get("pk")
+        floors = list(
+            self.queryset.filter(location__id=location_id)
+            .values_list("floorplan__floor", flat=True)
+            .distinct()
+        )
+        return floors
```

Comment:
I had some reservations on how this will handle the multi-tenant support, i.e. an administrator of one org tries to fetch the data using location_id of another org.

While testing this, I found that this endpoint does not work for non-superuser. 


![Screenshot from 2025-07-08 16-57-43](https://github.com/user-attachments/assets/20f4258c-717a-43c9-8d22-c17d7ad95c73)
![Screenshot from 2025-07-08 16-57-56](https://github.com/user-attachments/assets/3646e730-ed3b-4cc0-8202-ceee7a766f54)

Refer this docs section in openwisp-users 
https://openwisp.io/docs/dev/users/developer/django-rest-framework-utils.html#checking-parent-objects